### PR TITLE
Allow developers to register contracts as service contracts

### PR DESCRIPTION
**Pull request by Buckram123**
See ABS-485
### Checklist

- [ ] CI is green.
- [x] Changelog updated.

### CODE REVIEW

File: framework/packages/abstract-client/src/account.rs [hunk 1]
```diff
@@ -927,30 +961,6 @@ impl<Chain: CwEnv> Account<Chain> {
...
-        // When there are multiple modules registered the addresses are returned in a common
-        // separated list. We want the last one as that is the "top-level" module while the rest
-        // are dependencies, since in the sub-account creation call, we pass in the top-level
-        // module last.
-        let module_address = module_addresses.split(',').last().unwrap();
-
-        // We install only one module
-        Addr::unchecked(module_address)
-    }
-
```
Comment (Buckram123) [inline framework/packages/abstract-client/src/account.rs:953]: This method is not useful in combination with removing duplicate modules from install list, so got replaced to the more reliable method

File: framework/docs/src/releases/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@
 ### Added
 
 - `execute_with_funds` to Executor to attach funds to execution.
+- New module type: `Service`, acts the same as Native, but can be registered by any namespace
```
Comment (CyberHoward) [inline framework/docs/src/releases/CHANGELOG.md:8]:
```suggestion
- New module type: `Service`, behaves the same as Native, but can be registered by any namespace.
```

File: framework/packages/abstract-client/src/account.rs [hunk 3]
```diff
@@ -521,6 +539,22 @@ impl<Chain: CwEnv> Account<Chain> {
         }
     }
 
+    /// Install an service on the account.
+    /// if `install_on_sub_account` is `true`, the application will be installed on new a sub-account. (default)
```
Comment (CyberHoward) [inline framework/packages/abstract-client/src/account.rs:543]: We disabled the sub-account install by default.
```suggestion
    /// if `install_on_sub_account` is `true`, the application will be installed on new a sub-account.
```

File: framework/packages/abstract-client/src/account.rs [hunk 4]
```diff
@@ -831,24 +865,24 @@ impl<Chain: CwEnv> Account<Chain> {
...
-            .manager
-            .install_modules(modules, Some(funds))?;
-
-        let module_addr = Self::parse_modules_installing_response(install_module_response);
         let contract = Contract::new(
             M::installed_module_contract_id(&self.id()?),
             self.environment(),
         );
-        contract.set_address(&module_addr);
+        contract.set_address(&module_addresses.modules[0].1);
```
Comment (CyberHoward) [inline framework/packages/abstract-client/src/account.rs:879]: is this guaranteed to be index 0 on this array?

Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs:879]: Yeah, we error when we get no modules
https://github.com/AbstractSDK/abstract/blob/6a6abf82ca05bdefc6f21f58de31bc7ffb302452/framework/contracts/account/manager/src/queries.rs#L187-L191

File: framework/packages/abstract-interface/src/deployers.rs [hunk 5]
```diff
@@ -242,3 +242,58 @@ pub trait StandaloneDeployer<Chain: CwEnv>: Sized + Uploadable + ContractInstanc
         Ok(())
     }
 }
+
+/// Trait for deploying Services
+pub trait ServiceDeployer<Chain: CwEnv>:
+    Sized + Uploadable + ContractInstance<Chain> + CwOrchInstantiate<Chain>
+{
+    /// Deploys the module. If the module is already deployed, it will return an error.
+    /// Use `maybe_deploy` if you want to deploy the module only if it is not already deployed.
```
Comment (CyberHoward) [inline framework/packages/abstract-interface/src/deployers.rs:251]: there is no `maybe_deploy` fn, that's what the strategies are for right?

**Review by CyberHoward — APPROVED** [review]

File: framework/packages/abstract-client/src/account.rs (hunk 4 above)
Comment (Buckram123) [reply framework/packages/abstract-client/src/account.rs]: Updated this query method on manager!

**Pull request merged**